protos/flyteidl/core/types.proto

```diff
@@ -98,3 +98,102 @@ message Error {
     // Error message thrown.
     string message = 2;
 }
+
+// Secret encapsulates information about the secret a task needs to proceed. An environment variable
+// _FLYTE_SECRET_ENV_PREFIX will be passed to indicate the prefix of the environment variables that will be present if
+// secrets are passed through environment variables.
+// _FLYTE_SECRET_PATH_PREFIX will be passed to indicate the prefix of the path where secrets will be mounted if secrets
+// are passed through file mounts.
+message Secret {
+    enum Type {
```

Comment:
why is this necessary?

Reply:
I mean is there a use to know the type of secret? can we actually handle this correctly